synapse/rest/client/sync.py

```diff
@@ -139,7 +139,28 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
             device_id,
         )
 
-        request_key = (user, timeout, since, filter_id, full_state, device_id)
+        # ID of the last ignored users account data event for this user,
+        # if we're initial syncing.
+        # We include this in the request key to invalidate an initial sync
+        # in the response cache once the set of ignored users has changed.
+        # (We filter out ignored users from timeline events, so our sync response
+        # is invalid once the set of ignored users changes.)
+        last_ignored_users_id: Optional[int] = None
```

Comment:
Naming a little unfortunate: it makes it sounds like this holds a user id(s) rather than a stream id. Not sure there's a better alternative though: `last_ignored_users_streampos`?